### PR TITLE
Ruby debugger integration

### LINKED ISSUE #3758: Integrate Ruby Debugger
Tasks - 
- [x] Add Rubby debugger `debug` gem & update scripts to enable debugger at runtime
- [x] Add configuration for debugging with VSCode
- [x] Add configuration for debugging with Chrome
- [x] Modiy docker configuration to support remote debugging
- [x] Add configuration for remote debugging with VSCode 
- [x] Update documentation

Are you working on this? (Yes/No)
Yes

### PR DESCRIPTION
**Pull request by tanmoysrt**
Fixes #3758 

#### Describe the changes you have made in this PR -
- [x] Add Rubby debugger `debug` gem & update scripts to enable debugger at runtime
- [x] Add configuration for debugging with VSCode
- [x] Add configuration for debugging with Chrome
- [x] Modiy docker configuration to support remote debugging
- [x] Add configuration for remote debugging with VSCode in docker based installation
- [x] Update documentation

#### Working of Ruby Debugger

**Debugging With VSCode [local Installation]**

https://github.com/CircuitVerse/CircuitVerse/assets/57363826/1c0e957e-3eed-4555-8670-27df282786ab

**Debugging With Chrome [local installation]**

https://github.com/CircuitVerse/CircuitVerse/assets/57363826/0d7e8313-602e-4a79-a127-8fae572da82f

**Debugging With VSCode [docker based Installation]**

https://github.com/CircuitVerse/CircuitVerse/assets/57363826/2fad0794-35a3-40f5-b863-f934a1cdab05

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

**Comment by tanmoysrt** [comment]
@vedant-jain03 @smritigarg kindly review this PR

File: Gemfile [hunk 1]
```diff
@@ -118,8 +118,8 @@ gem "strong_migrations"
 gem 'rails-data-migrations'
 
 group :development, :test do
-  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
-  gem "byebug", platforms: %i[mri mingw x64_mingw]
+  # Adds support for debug
+  gem "debug", "~>1.8"
```
Comment (smritigarg) [inline Gemfile:122]: **Question** Any reason for locking gem to a particular version ?

File: bin/dev [hunk 2]
```diff
@@ -6,4 +6,10 @@ then
   gem install foreman
 fi
 
-foreman start -f Procfile.dev
+if [[ "$1" == "chrome_debug" ]]; then
```
Comment (smritigarg) [inline bin/dev:9]: **Question** Are we sure this value will not change in future `chrome_debug` shall we store it in an env variable instead?

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```
Comment (smritigarg) [inline .vscode/launch.json:4]: **Suggestion** I think we can get rid of these comments. What say?

**Review by smritigarg — COMMENTED** [review]
Hi, @tanmoysrt Great Job :D !! 

I did first round of interview and had some queries, My local setup is giving some issues and I am not able to execute and check your changes yet. But so far so good 😄  !!

File: Gemfile (hunk 1 above)
Comment (tanmoysrt) [reply Gemfile:122]: Not any reason. It's the latest version. 
Will remove if required

File: bin/dev (hunk 2 above)
Comment (tanmoysrt) [reply bin/dev:9]: It's part of the bash script customized for the purpose of using chrome based debugger

- If we run `./bin/dev` that will run solargraph and we can connect from vscode or other supported IDE
  -  in this case that will use Procfile.dev
- If we run `./bin/dev chrome_debug` that will open up chrome for debugging.
 - In this case that will use Porcfile.chrome.dev

To just choosing, which Procfile should be use chrome_debug flag has been added

File: .vscode/launch.json (hunk 3 above)
Comment (tanmoysrt) [reply .vscode/launch.json]: Sure
Removed !

File: bin/dev (hunk 2 above)
Comment (smritigarg) [reply bin/dev:9]: Sounds great lets go ahead with this

File: bin/rdbg [hunk 4]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+#
+# This file was generated by Bundler.
+#
+# The application 'rdbg' is installed as part of a gem, and
+# this file is here to facilitate running it.
+#
```
Comment (smritigarg) [inline bin/rdbg:9]: Please get rid of these comments

**Review by vedant-jain03 — APPROVED** [review]
LGTM 🚀 
cc @tachyons

**Comment by tachyons** [comment]
Looks great, could you also add documentation for neovim and sublime in a follow-up pull request?

**Pull request merged**